app/src/main/kotlin/com/camerasecuritysystem/client/ConnectDialog.kt

```diff
@@ -242,11 +249,18 @@ class ConnectDialog : AppCompatDialogFragment() {
         val ipAddress_til = binding.inputLayoutIpAddress
 
         if (ipAddressText == null || ipAddressText.isEmpty()) {
-            ipAddress_til.error = String.format(resources.getString(R.string.err_not_empty), ipaddressStr)
+            ipAddress_til.error =
+                String.format(resources.getString(R.string.err_not_empty), ipaddressStr)
+            return false
+        }
+        if (!Pattern.compile(
+                "((^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\$)|" +
+                        "^(^(?!\\bw{3}\\.\\b)[a-zA-Z0-9-_]{0,61}([a-zA-Z0-9])" +
+                        ".[a-zA-Z0-9-_]{1,})\\.[a-z]{2,4}\$)",
+                Pattern.CASE_INSENSITIVE).matcher(ipAddressText).find()) {
```

Comment:
Maybe this regex could make use of a comment describing what it checks for? Doesn't have to be the most detailed comment. 

But it checks for an ipv4 address containing 4 digits ranging from 1 to 999 separated by dots.

The domain check allows base domain names starting from 1 characters to "unlimited". When a subdomain is used it has to be in the range of 1 to 63 characters.

We should also be aware that this check is not a bullet proof compliance check with the domain naming standard and should only be used to help the user out, not to protect the input against non compliant domains/ipv's.